source/MaterialXTest/ShaderGen.cpp

```diff
@@ -704,28 +705,26 @@ TEST_CASE("ShaderX Implementation Validity", "[shadergen]")
     implDumpBuffer.open(fileName, std::ios::out);
     std::ostream implDumpStream(&implDumpBuffer);
 
-    // Node types to explicitly skip temporarily.
-    std::list<std::string> skipNodeTypes;
-    skipNodeTypes.push_back("ambientocclusion");
-    skipNodeTypes.push_back("arrayappend");
-    skipNodeTypes.push_back("blur");
-    skipNodeTypes.push_back("curveadjust");
-
     for (auto generator : shaderGenerators)
     {
         generator->registerSourceCodeSearchPath(searchPath);
 
         const std::string& language = generator->getLanguage();
         const std::string& target = generator->getTarget();
 
+        // Node types to explicitly skip temporarily.
+        std::set<std::string> skipNodeTypes = 
+        {
+            "ambientocclusion",
+            "arrayappend",
+            "blur",
+            "curveadjust",
+        };
+
         // Skip lights for OSL for now
         if (language == "sx-osl")
         {
-            skipNodeTypes.push_back("light");
-        }
-        else
-        {
-            skipNodeTypes.remove("light");
+            skipNodeTypes.insert("light");
         }
```

Comment:
The remove still needs to be done otherwise light checking will removed when the glsl is tested.